modules/videoeditor-ffmpeg-impl/src/main/java/org/opencastproject/videoeditor/impl/VideoEditorServiceImpl.java

```diff
@@ -279,19 +278,88 @@ protected Track processSmil(Job job, Smil smil, String trackParamGroupId) throws
+                + "There likely is an error in the SMIL file");
+      }
+
+      // get output file extension
+      String outputFileExtension = null;
+      if (videoclips.size() > 0) {
+        outputFileExtension = properties.getProperty(VideoEditorProperties.DEFAULT_EXTENSION, ".mp4");
+      }
+      if (refElements.size() > 0) {
+        String extension = FilenameUtils.getExtension(sourceTrackUri);
+        if (VideoEditorProperties.WEBVTT_EXTENSION.equals(extension)) {
+          outputFileExtension = properties.getProperty(VideoEditorProperties.WEBVTT_EXTENSION, ".vtt");
+        }
+      }
+      outputFileExtension = properties.getProperty(VideoEditorProperties.OUTPUT_FILE_EXTENSION, outputFileExtension);
+
+      if (!outputFileExtension.startsWith(".")) {
+        outputFileExtension = '.' + outputFileExtension;
+      }
+
+      String filename = String.format("%s-%s%s", sourceTrackFlavor,
+              FilenameUtils.removeExtension(sourceFile.getName()), outputFileExtension);
+      File outputPath = new File(tempDirectory, filename);
+
+      if (!outputPath.getParentFile().exists()) {
+        outputPath.getParentFile().mkdirs();
+      }
+
+      // If we are cutting video/audio, use ffmpeg
+      if (videoclips.size() > 0) {
+        List<VideoClip> cleanclips = sortSegments(videoclips);    // remove very short cuts that will look bad
+        String error = null;
+        String outputResolution = "";    //TODO: fetch the largest output resolution from SMIL.head.layout.root-layout
+        // When outputResolution is set to WxH, all clips are scaled to that size in the output video.
+        // TODO: Each clips could have a region id, relative to the root-layout
+        // Then each clip is zoomed/panned/padded to WxH before concatenation
+        FFmpegEdit ffmpeg = new FFmpegEdit(properties);
+        error = ffmpeg.processEdits(inputfile, outputPath.getAbsolutePath(), outputResolution, cleanclips,
+                sourceTrack.hasAudio(), sourceTrack.hasVideo());
+
+        if (error != null) {
+          FileUtils.deleteQuietly(tempDirectory);
+          throw new ProcessFailedException("Editing pipeline exited abnormally! Error: " + error);
+        }
+      }
+
+      // If we are cutting ref elements, check if they are subtitle files
+      // Or give up
+      // TODO: It might be better if subtitle tracks were assigned the mediatype "texttrack" in the first place
+      if (refElements.size() > 0) {
+        List<VideoClip> cleanclips = sortSegments(refElements);    // remove very short cuts that will look bad
+        String extension = FilenameUtils.getExtension(sourceTrackUri);
+        if (VideoEditorProperties.WEBVTT_EXTENSION.equals(extension)) {
+          // Parse
+          WebVTTParser parser = new WebVTTParser();
+          FileInputStream fin = new FileInputStream(sourceFile);
+          WebVTTSubtitle subtitle = parser.parse(fin);
```

Comment:
Close the input stream:
```suggestion
          WebVTTSubtitle subtitle;
          try (FileInputStream fin = new FileInputStream(sourceFile)) {
            subtitle = parser.parse(fin);
          }
```